cmd/kubelet/app/server.go

```diff
@@ -201,6 +201,7 @@ is checked every 20 seconds (also configurable with a flag).`,
 
 			if cleanFlagSet.Changed("pod-infra-container-image") {
 				klog.InfoS("--pod-infra-container-image will not be pruned by the image garbage collector in kubelet and should also be set in the remote runtime")
+				_ = cmd.Flags().MarkDeprecated("pod-infra-container-image", "--pod-infra-container-image will be removed in 1.30. Image garbage collector will get sandbox image information from CRI.")
```

Comment:
this flag cannot be removed one release after it is deprecated.

Reply:
see https://github.com/kubernetes/kubeadm/issues/2626#issuecomment-1763244783

Reply:
The flag is deprecated twice, here it is deprecated every time it is changed (?)

cmd/kubelet/app/server.go:                              _ = cmd.Flags().MarkDeprecated("pod-infra-container-image", "--pod-infra-container-image will be removed in 1.30. Image garbage collector will get sandbox image information from CRI.")

It was already deprecated, with a different message, here: Originally: "in 1.27"

pkg/kubelet/config/flags.go:    fs.MarkDeprecated("pod-infra-container-image", "will be removed in a future release. Image garbage collector will get sandbox image information from CRI.")

Reply:
yes, i know it was deprecation before by mistake. the problem is that before 1.29 there is no workaround for the GC problem. IMO, this means the flag served a functionality that falls under the GA deprecation policy.